config/winston.js

```diff
@@ -0,0 +1,71 @@
+var appRoot = require('app-root-path')
+var winston = require('winston')
+
+const { combine, colorize, printf, timestamp } = winston.format
+
+const logFormat = printf((info) => {
+  return `[${info.timestamp}] ${info.level}: ${info.message}`
+})
+
+const rawFormat = printf((info) => {
+  return `[${info.timestamp}] ${info.level}: ${info.message}`
+})
+
+// define the custom settings for each transport (file, console)
+var options = {
+  file: {
+    level: 'info',
+    filename: `${appRoot}/logs/app.log`,
+    handleExceptions: true,
+    json: true,
+    maxsize: 5242880, // 5MB
+    maxFiles: 5,
+    colorize: false
+  },
+  errorFile: {
+    level: 'error',
+    name: 'file.error',
+    filename: `${appRoot}/logs/error.log`,
+    handleExceptions: true,
+    json: true,
+    maxsize: 5242880, // 5MB
+    maxFiles: 100,
+    colorize: true
+  },
+  console: {
+    level: 'debug',
+    handleExceptions: true,
+    json: false,
+    format: combine(colorize(), rawFormat)
+  }
+}
+
+// instantiate a new Winston Logger with the settings defined above
+var logger = winston.createLogger({
+  format: combine(
+    timestamp({ format: 'YYYY-MM-DD HH:mm:ss' }),
+    logFormat
+  ),
+  transports: [
+    new winston.transports.File(options.file),
+    new winston.transports.Console(options.console)
+  ],
+  exitOnError: false // do not exit on handled exceptions
+})
+
+// create a stream object with a 'write' function that will be used by `morgan`
+logger.stream = {
+  write: function (message, encoding) {
+    // use the 'info' log level so the output will be picked up by both transports (file and console)
+    logger.info(message)
```

Comment:
@Rupeshiya by doing this we will be logging info in console even if we dont want to.
Can we modify this such that in development environment we prints logs on browser console as well as server side logs and in prod env we do it just on the server side.

Reply:
Yes, @devesh-verma I have already done that part for production in frontend part like in prod there would not be anything in the browser console.

And If we want to have log both in development and production in server-side then why we need to check env??

Reply:
These are not affecting browser console in anyway.

Reply:
not on server-side, I was talking with respect to client-side. This is good then 👍